File: code_templates/agent.py
from config_manager import ConfigManager
from memory import Memory
from tools import Tools
from task_parser import TaskParser

class Agent:
    def __init__(self):
        self.config = ConfigManager()
        self.memory = Memory()
        self.tools = Tools()
        self.parser = TaskParser()

    def handle(self, user_input):
        task = self.parser.parse(user_input)

        if task["type"] == "search":
            result = self.tools.search_web(task["query"])
        elif task["type"] == "email":
            result = self.tools.send_email(task["to"], task["subject"], task["body"])
        else:
            result = f"回覆: {task['text']}"

        self.memory.add(user_input, result)
        return result

File: code_templates/agent.ts
// Agent.ts
import { ConfigManager } from "./ConfigManager";
import { Memory } from "./Memory";
import { Tools } from "./Tools";
import { TaskParser } from "./TaskParser";

export class Agent {
  private config: ConfigManager;
  private memory: Memory;
  private tools: Tools;
  private parser: TaskParser;

  constructor() {
    this.config = new ConfigManager();
    this.memory = new Memory();
    this.tools = new Tools();
    this.parser = new TaskParser();
  }

  handle(userInput: string): string {
    const task = this.parser.parse(userInput);
    let result: string;

    switch (task.type) {
      case "search":
        result = this.tools.searchWeb(task.query);
        break;
      case "email":
        result = this.tools.sendEmail(task.to, task.subject, task.body);
        break;
      default:
        result = `回覆: ${task.text}`;
    }

    this.memory.add(userInput, result);
    return result;
  }
}
